modules/navier_stokes/test/tests/finite_element/ins/energy-conservation/q1q1.i

```diff
@@ -0,0 +1,184 @@
+[Mesh]
+  [gen]
+    type = GeneratedMeshGenerator
+    nx = 10
+    ny = 10
+    dim = 2
+  []
+  [subdomain]
+    type = SubdomainBoundingBoxGenerator
+    bottom_left = '0.5 0 0'
+    top_right = '1 1 0'
+    block_id = 1
+    input = gen
+  []
+  [break_boundary]
+    input = subdomain
+    type = BreakBoundaryOnSubdomainGenerator
+    boundaries = 'bottom top'
+  []
+  [sideset]
+    type = SideSetsBetweenSubdomainsGenerator
+    input = break_boundary
+    primary_block = '1'
+    paired_block = '0'
+    new_boundary = 'fluid_left'
+  []
+  coord_type = RZ
+[]
+
+[Variables]
+  [T][]
+  [velocity]
+    family = LAGRANGE_VEC
+    block = 1
+  []
+  [pressure]
+    block = 1
+  []
+[]
+
+[ICs]
+  [vel]
+    type = VectorConstantIC
+    variable = velocity
+    x_value = 1e-15
+    y_value = 1e-15
+    block = 1
+  []
+[]
+
+[Kernels]
+  [mass]
+    type = INSADMass
+    variable = pressure
+    block = 1
+  []
+  [pspg]
+    type = INSADMassPSPG
+    variable = pressure
+    block = 1
+  []
+
+  [momentum_convection]
+    type = INSADMomentumAdvection
+    variable = velocity
+    block = 1
+  []
+  [momentum_viscous]
+    type = INSADMomentumViscous
+    variable = velocity
+    block = 1
+  []
+  [momentum_pressure]
+    type = INSADMomentumPressure
+    variable = velocity
+    pressure = pressure
+    integrate_p_by_parts = true
+    block = 1
+  []
+  [momentum_supg]
+    type = INSADMomentumSUPG
+    variable = velocity
+    velocity = velocity
+    block = 1
+  []
+
+  [temperature_advection]
+    type = INSADEnergyAdvection
+    variable = T
+     block = 1
+  []
+  [temperature_supg]
+    type = INSADEnergySUPG
+    variable = T
+    velocity = velocity
+    block = 1
+  []
+  [temperature_conduction]
+    type = ADHeatConduction
+    variable = T
+    thermal_conductivity = 'k'
+  []
+  [heat_source]
+    type = BodyForce
+    variable = T
+    block = 0
+    function = 'x + y'
+  []
+[]
+
+[BCs]
+  [velocity_inlet]
+    type = VectorFunctionDirichletBC
+    variable = velocity
+    function_y = 1
+    boundary = 'bottom_to_1'
+  []
+  [wall]
+    type = VectorFunctionDirichletBC
+    variable = velocity
+    boundary = 'fluid_left right'
+  []
+  [convective_heat_transfer]
+    type = ConvectiveHeatFluxBC
+    variable = T
+    T_infinity = 0
```

Comment:
oh yeah this should remove the null space?

Reply:
even though it s a neumann BC right

Reply:
yessir